[ao][sparsity] adding type hints to sparsifier and utils

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79774
* __->__ #79598
* #79597

Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity] adding type hints to sparsifier and utils

Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

z-a-f commented on torch/ao/sparsity/sparsifier/base_sparsifier.py:
```diff
@@ -1,21 +1,27 @@
-KEYS_NOT_IN_STATE_DICT = ['module', 'module_fqn', 'tensor_name']
+KEYS_NOT_IN_STATE_DICT = ["module", "module_fqn", "tensor_name"]
+
+__all__ = ["BaseSparsifier"]
```
dup?

HDCharles replied:
pyfmt prefers that i think

z-a-f commented on torch/ao/sparsity/sparsifier/base_sparsifier.py:
```diff
@@ -42,41 +48,41 @@ class BaseSparsifier(abc.ABC):
-        self.defaults = defaults
-        if self.defaults is None:
-            self.defaults = dict()
+        if defaults is None:
+            defaults = dict()
+        self.defaults: Dict[str, Any] = defaults
```
nit: 
```
self.defaults: Dict[str, Any] = defaults or dict()
```

HDCharles replied:
that doesn't seem to cause an issue actually

HDCharles committed:
Update on "[ao][sparsity] adding type hints to sparsifier and utils"


Summary: no functional changes to the code, only type hints/formatting

Test Plan: python test/test_ao_sparsity.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
